core/src/main/java/com/adobe/aem/commons/assetshare/content/renditions/impl/AssetRenditionServlet.java

```diff
@@ -0,0 +1,157 @@
+
+    @Reference
+    private AssetRenditions assetRenditions;
+
+    public final void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response) throws IOException, ServletException {
+        final AssetRenditionDispatcher.Params params = new ParamsImpl(request);
+
+        if (params.isValid()) {
+            for (final AssetRenditionDispatcher assetRenditionDispatcher : assetRenditions.getAssetRenditionDispatchers()) {
+                if (assetRenditionDispatcher.accepts(request, params.getRenditionName())) {
+
+                    setResponseHeaders(response, params);
+
+                    assetRenditionDispatcher.dispatch(request, response);
+                    return;
+                }
+            }
+        } else {
+            log.debug("Request suffix [ {} ] has invalid 'suffix extension'", request.getRequestPathInfo().getSuffix());
+        }
+
+        response.sendError(HttpServletResponse.SC_NOT_FOUND,
+                "Unable to locate a AssetRenditionDispatcher to dispatch a rendition for [" + params.getRenditionName() + "].");
+    }
+
+    protected void setResponseHeaders(final SlingHttpServletResponse response, final AssetRenditionDispatcher.Params params) {
+        if (params.isAttachment()) {
+            response.setHeader("Content-Disposition", String.format("attachment; filename=%s", params.getFileName()));
+        } else {
+            response.setHeader("Content-Disposition", String.format("filename=%s", params.getFileName()));
+        }
+    }
+
+    /**
+     * Represents the parameters provided in the RequestPathInfo's suffix to determine how the rendition is selected and returned.
+     */
+    protected static class ParamsImpl implements AssetRenditionDispatcher.Params {
+        private String renditionName = null;
+        private String fileName = null;
+        private boolean attachment = false;
+        private boolean valid = true;
+
+        public ParamsImpl(SlingHttpServletRequest request) {
+            final Asset asset = DamUtil.resolveToAsset(request.getResource());
+            final String[] segments = StringUtils.split(request.getRequestPathInfo().getSuffix(), "/");
+
+            if (asset == null ||
+                    segments.length < 2 ||
+                    (!CACHEABLE_SUFFIX_FILENAME.equals(segments[segments.length - 1]) &&
+                            !StringUtils.startsWith(segments[segments.length - 1], CACHEABLE_SUFFIX_FILENAME + "."))) {
+                valid = false;
+            } else {
+                if (segments.length > 0) {
+                    renditionName = StringUtils.stripToNull(StringUtils.substringBefore(segments[0], "."));
+                }
+
+                attachment = ArrayUtils.indexOf(segments, DOWNLOAD_AS_ATTACHMENT_SUFFIX_SEGMENT) > 0;
+
+                final int dotIndex = StringUtils.lastIndexOf(asset.getName(), ".");
+
+                if (dotIndex < 0) {
+                    fileName = asset.getName() + "." + renditionName;
```

Comment:
I guess it should throw a 500? The alternative is to make the Servlet and opting servlet that ensures its got a renditionName and the cacheable last suffix..?

Reply:
@bstopp  wrt to the "asset.rendition" as the last suffix... we need something with an "extension" (ie: DOT something) as the last suffix segment for caching. I was trying to figure out a clean way to set an "appropraite" extension, but it turns out its actually hard to know what extension something is going to result to....

for example, you could very well have ... 

`/content/dam/test.png/renditions/my-custom/asset.rendition` serve a PNG.. or maybe a JPG (or really anything, and it could change over time). The Content-Type is (should be) set on the response header indicating the browser the type of content on the response.

This gets even harder if we create something like a DynamicMedia Dispatcher, that 301/302's out to something else. If we do require a mimType'y extension (`/content/dam/test.png/renditions/my-custom/asset.png`) what happens when (over time perhaps) content on the resposne doesnt match that extension?

I guess, are there any major downsides of specifying the Content-Type via response headers and NOT the extension? The main one i can think of is HTTPD ModMimeTypeInfo which IIRC goes off the extension?